lib/ansible/modules/cloud/azure/azure_rm_managed_disk.py

```diff
@@ -231,25 +250,71 @@ def exec_module(self, **kwargs):
         """Main module execution method"""
         for key in list(self.module_arg_spec.keys()) + ['tags']:
             setattr(self, key, kwargs[key])
-        results = dict()
-        resource_group = None
-        response = None
-        try:
-            resource_group = self.get_resource_group(self.resource_group)
-        except CloudError:
-            self.fail(
-                'resource group {} not found'
-                .format(self.resource_group))
+        result = None
+        resource_group = self.get_resource_group(self.resource_group)
         if not self.location:
             self.location = resource_group.location
+
+        result = self.get_managed_disk()
         if self.state == 'present':
-            self.results['state'] = self.create_or_update_managed_disk()
-        elif self.state == 'absent':
-            self.delete_managed_disk()
+            result, changed = self.create_or_update_managed_disk(result)
+
+        # unmount from the old virtual machine and mount to the new virtual machine
+        if result:
```

Comment:
This should probably be in the same if statement as `if self.state == 'present'`, this way you know the disk exists and so the initial present check is not required and you never really want to run this additional operation if `state: absent`.

Reply:
if the state is absent, and the managed_by is None, the logic should be unmount the disk first and then delete the disk. Otherwise, Azure API will throw error "the disk is managed by XXX, so you cannot delete it blablablah"